File: Binary Tree/111. Minimum Depth of Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
#include <limits.h>
int mindepth = INT_MAX;

int getmin(struct TreeNode *root, int current)
{
    if (root == NULL)
    {
        return 0;
    }

    int left = getmin(root->left, current + 1);
    int right = getmin(root->right, current + 1);

    if (left == 0 && right == 0)
    {
        if (current < mindepth)
        {
            mindepth = current;
        }
    }

    return 1;
}

int minDepth(struct TreeNode *root)
{
    if (root == NULL)
    {
        return 0;
    }
    mindepth = INT_MAX;

    int min = getmin(root, 1);

    return mindepth;
}

File: Graphs/111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        res = float('inf')
        if not root:
            return 0
        q = deque()
        q.append([root, 1])
        while q:
            f, h = q.popleft()
            if not f.left and not f.right:
                res = min(res, h)
            if f.left:
                q.append([f.left, h + 1])
            if f.right:
                q.append([f.right, h + 1])
        return res